testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCoercionOnPartitionedTable.java

```diff
@@ -142,6 +142,9 @@ private static HiveTableDefinition.HiveTableDefinitionBuilder tableDefinitionBui
                         "    varchar_to_smaller_varchar         VARCHAR(3)," +
                         "    varchar_to_date                    VARCHAR(10)," +
                         "    varchar_to_distant_date            VARCHAR(12)," +
+                        "    varchar_to_double                  VARCHAR(40)," +
+                        "    varchar_to_double_infinity         VARCHAR(40)," +
+                        "    varchar_to_special_double          VARCHAR(40)," +
```

Comment:
Can we have one of the column as `String` and `unBoundedVarchar` and update the column names accordingly ?

Reply:
Added `string_to_double` column to add coverage for the coercion to double of unbouned varchar columns .